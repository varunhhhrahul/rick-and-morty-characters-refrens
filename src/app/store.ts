import { configureStore, Action, getDefaultMiddleware } from "@reduxjs/toolkit";
import { ThunkAction } from "redux-thunk";

import rootReducer, { RootState } from "./rootReducer";

const store = configureStore({
  reducer: rootReducer,
  middleware: () => [...getDefaultMiddleware()],
  // middleware: [...getDefaultMiddleware({ immutableCheck: false })],
});

export type AppDispatch = typeof store.dispatch;

export type AppThunk = ThunkAction<void, RootState, null, Action<string>>;

export default store;
