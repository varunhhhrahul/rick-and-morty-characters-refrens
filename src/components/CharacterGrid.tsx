import React, { useEffect } from "react";
import { shallowEqual, useDispatch, useSelector } from "react-redux";
import { RootState } from "../app/rootReducer";
import { getCharactersForPage } from "../slices/characterSlice";

interface CharacterGridProps {}

export const CharacterGrid: React.FC<CharacterGridProps> = ({}) => {
  const { characters, page, nextPage } = useSelector((state: RootState) => {
    return {
      characters: state.character.characters,
      page: state.character.page,
      nextPage: state.character.nextPage,
    };
  }, shallowEqual);
  const dispatch = useDispatch();
  useEffect(() => {
    // dispatch(getCharactersForPage(page));
    dispatch(getCharactersForPage(page));
  }, [dispatch, page]);

  return <div></div>;
};
