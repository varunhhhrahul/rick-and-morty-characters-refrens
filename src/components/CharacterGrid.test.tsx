import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../app/store";
import { CharacterGrid } from "./CharacterGrid";

describe("CharacterGrid", () => {
  test("should render correctly", async () => {
    const component = renderer.create(
      <Provider store={store}>
        <CharacterGrid />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
