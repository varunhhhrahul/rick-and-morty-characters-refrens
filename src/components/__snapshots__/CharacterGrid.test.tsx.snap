// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterGrid should render correctly 1`] = `
<div
  className="MuiBox-root css-16dwg1o"
>
  <h5
    className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiTypography-gutterBottom css-anaryy-MuiTypography-root"
  >
    Rick and Morty Characters
  </h5>
  <div
    className="MuiFormControl-root MuiTextField-root css-136h9lv-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor="outlined-password-input"
      id="outlined-password-input-label"
    >
      Search
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="outlined-password-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Search for a character"
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1in441m"
        >
          <span>
            Search
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="css-bvx1az-MuiStack-root"
  >
    <div>
      <nav
        aria-label="pagination navigation"
        className="MuiPagination-root MuiPagination-outlined css-1oj2twp-MuiPagination-root"
      >
        <ul
          className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-previousNext css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-current="true"
              aria-label="page 1"
              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-selected MuiPaginationItem-page css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              1
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-outlinedPrimary Mui-disabled MuiPaginationItem-previousNext css-r93niq-MuiButtonBase-root-MuiPaginationItem-root"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="MuiGrid-root css-1gptvh3-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5abhjx-MuiPaper-root-MuiCard-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1slz293-MuiSkeleton-root"
          style={
            Object {
              "height": 800,
              "width": 300,
            }
          }
        />
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5abhjx-MuiPaper-root-MuiCard-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1slz293-MuiSkeleton-root"
          style={
            Object {
              "height": 800,
              "width": 300,
            }
          }
        />
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5abhjx-MuiPaper-root-MuiCard-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1slz293-MuiSkeleton-root"
          style={
            Object {
              "height": 800,
              "width": 300,
            }
          }
        />
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5abhjx-MuiPaper-root-MuiCard-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1slz293-MuiSkeleton-root"
          style={
            Object {
              "height": 800,
              "width": 300,
            }
          }
        />
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5abhjx-MuiPaper-root-MuiCard-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1slz293-MuiSkeleton-root"
          style={
            Object {
              "height": 800,
              "width": 300,
            }
          }
        />
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5abhjx-MuiPaper-root-MuiCard-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1slz293-MuiSkeleton-root"
          style={
            Object {
              "height": 800,
              "width": 300,
            }
          }
        />
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5abhjx-MuiPaper-root-MuiCard-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1slz293-MuiSkeleton-root"
          style={
            Object {
              "height": 800,
              "width": 300,
            }
          }
        />
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5abhjx-MuiPaper-root-MuiCard-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1slz293-MuiSkeleton-root"
          style={
            Object {
              "height": 800,
              "width": 300,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
